### PR TITLE
Fix #37 and add possibility to use UniTasks in Edit Mode

### LINKED ISSUE #37: Unitasks don't work properly in edit mode
I've tried to use Unitask into a "edit mode only" project (the unity project only creates assetbundles automatically), and UniTask runners and continuations don't work like play mode.
This example show the issue

```
public class UniTastEditModeIssue : MonoBehaviour
{
    [MenuItem("executeHeavyOperationByAsync", menuItem = "EditorTest/Execute Task")]
    static async Task VeryHeavyOpTask()
    {
        Debug.Log("Very Heavy Operation start (Task)");
        await Task.Delay(TimeSpan.FromSeconds(5));
        Debug.Log("Very Heavy Operation end (Task)");
    }
    
    [MenuItem("executeHeavyOperationbyUniTask", menuItem = "EditorTest/Execute UniTask")]
    static async UniTask VeryHeavyOpUniTask()
    {
        Debug.Log("Very Heavy Operation start (UniTask)");
        await UniTask.Delay(TimeSpan.FromSeconds(5));
        Debug.Log("Very Heavy Operation end (UniTask)");
    }
}
```

as you can see, the c# native Task works well, the UniTask version instead only print the first Debug.Log

After some research, i found the problem in PlayerLoop updates called in edit mode, and i found a solution using

```EditorApplication.QueuePlayerLoopUpdate();```

which forces unity to execute PlayerLoop lifecycle

i've made a pull request with my solution for the issue

(i'm sorry for any inadvertency, it's my first issue and pull request)

### PR DESCRIPTION
**Pull request by BattlefieldNoob**
Fix for #37, as stated by Thaina [Here](https://github.com/Cysharp/UniTask/issues/22#issuecomment-517126794), i've used `EditorApplication.playModeStateChanged` and implemented Clears to prevent runners and continuations to persist on play-to-edit or edit-to-play states change.

I've duplicated the Play mode unit test to use in Edit mode, i've saw that 3 unit test doesn't pass, this happen because the first MoveNext() of Promise is executed a frame after OnRunningStart(), i think this pull request can't be merged now.

### CODE REVIEW

**Comment by chanon** [comment]
I had some issues with UniTasks in Edit Mode (strange errors) .. and it seems this pull request fixed it for me. Thank you! And I support merging it.

**Comment by kamyker** [comment]
'using FluentAssertions;' ? I think you shouldn't use it in example test.

File: Assets/UniRx.Async/PlayerLoopHelper.cs [hunk 1]
```diff
@@ -62,38 +66,30 @@ public static class PlayerLoopHelper
         static ContinuationQueue[] yielders;
         static PlayerLoopRunner[] runners;
 
-        static PlayerLoopSystem[] InsertRunner(PlayerLoopSystem loopSystem, Type loopRunnerYieldType, ContinuationQueue cq, Type loopRunnerType, PlayerLoopRunner runner)
+        static PlayerLoopSystem[] InsertRunner(PlayerLoopSystem loopSystem, Type loopRunnerYieldType,
+            ContinuationQueue cq, Type loopRunnerType, PlayerLoopRunner runner)
         {
+            EditorApplication.playModeStateChanged += (state) =>
```
Comment (igor84) [inline Assets/UniRx.Async/PlayerLoopHelper.cs]: This part also has to be under `#if UNITY_EDITOR`

Comment (BattlefieldNoob) [reply Assets/UniRx.Async/PlayerLoopHelper.cs]: It's right, thank you, i've updated the pull request

**Comment by BattlefieldNoob** [comment]
Thank you guys for your support, as stated in my first comment unfortunately not all editor test pass, i ask you some help to fix that, but i think that the only solution is to modify them (because in editor time Unity change the execution order of some callbacks)

**Comment by JeffersonFerreira** [comment]
Hello guys.
Any news about this feature? :)

**Comment by neuecc** [comment]
I don't completely agree with the functionality and implementation, but
All right, let's put it in first.

**Comment by neuecc** [comment]
and thanks to nice implementation.

**Pull request merged**